packages/riverpod_lint/lib/src/assists/convert_to_widget_utils.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:analyzer/dart/ast/ast.dart';
+// ignore: implementation_imports, somehow not exported by analyzer
+import 'package:collection/collection.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+import '../object_utils.dart';
+
+enum StatelessBaseWidgetType {
+  consumerWidget(
+    widgetName: 'ConsumerWidget',
+
+    // But the priority above everything else
+    priority: 100,
+    typeChecker: TypeChecker.fromName(
+      'ConsumerWidget',
+      packageName: 'flutter_riverpod',
+    ),
+  ),
+  hookWidget(
+    widgetName: 'HookWidget',
+    priority: 98,
+    typeChecker: TypeChecker.fromName(
+      'HookWidget',
+      packageName: 'flutter_hooks',
+    ),
+  ),
+  hookConsumerWidget(
+    widgetName: 'HookConsumerWidget',
```

Comment:
I'd expect hook variants to have a higher priority than non-hook variants, but only show up if the analyzed package depends on hooks_riverpod/fluter_hooks.

Reply:
Bump on this, how that hooks are optional :)

Reply:
I tried to change the priority order as follows:

1. ConvertToHookConsumerWidget
2. ConvertToHookWidget
3. ConvertToConsumerWidget
4. ConvertToStatelessWidget
5. ConvertToStatefulHookConsumerWidget
6. ConvertToStatefulHookWidget
7. ConvertToConsumerStatefulWidget
8. ConvertToStatefulWidget

However, I'm not very confident about this order, so I'd appreciate it if you could let me know if this order seems good. 

Additionally, I made some conversions optional.

Display only when there is a dependency on hooks_riverpod
- ConvertToHookConsumerWidget
- ConvertToStatefulHookConsumerWidget

Display only when there is a dependency on flutter_hooks
- ConvertToHookWidget
- ConvertToStatefulHookWidget